### DISCUSSION TITLE
Adding shortcut keys to perform certain actions like Build , Pin/Unpin editor, Close/Open editor,  Run and Reset

### DISCUSSION THREAD

**Discussion by surajkumar-sk** [sugarlabs/musicblocks-v4#185] [Ideas]
I am a Student participating in GSOC This year 2022. While learning about what different syntax does in musicblock v4 I had to edit and run the code a lot of times, I kept resetting the block and building the code for almost every 5 sec. It was fine moving my mouse to click the buttons in the beginning but after few attempts, It started getting frustrating to hover and click on build,  run and reset. I as a Developer will probably not build, run and reset many times. Buts kids and  teachers using this website might have to click on buttons more often. Adding shortcuts to buttons would make it easier to perform actions. we could add a hover text on buttons showing the shortcut keys. Changes in code wouldn't be much but adding this can make accessibility of buttons a lot easier for frequent learners or teachers.

shortcut for run - alt + r
shortcut for stop - alt + s
shortcut for reset - ctrl + alt + r
shortcut for opening or closing editor - ctrl + alt + e
shortcut for pin or unpin - alt + p
shortcut for build - alt + b
shortcut for help - alt + h

I would like to implement this changes as my `good first issue` and begin contributing to the project. 

Is this a good idea to implement? @meganindya

**Comment by meganindya** [comment]
The editor is in prototype stage right now. There is a prospective GSoC project idea to implement a powerful text editor. However, you can go ahead and implement this in the current editor. Please discuss your implementation idea ahead of coding though.

**Comment by jwalapc** [comment]
I think we can work on this.

**Comment by surajkumar-sk** [comment]
Yes , we can. I am just a bit busy with my sem exams and will start working as soon as i am free .

Feel free to come up with ideas on how to proceed I'll join as soon as my sem exams get finished.
